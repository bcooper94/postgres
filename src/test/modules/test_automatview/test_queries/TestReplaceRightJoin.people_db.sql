SET automatview.training_sample_count = 2;

select employee.id, person_id, employee_id from employee join manager on employee.id = employee_id;
select employee.id, person_id from employee join manager on employee.id = employee_id;

select person_id, name, age from person join employee on person.id = person_id join manager on employee.id = employee_id;