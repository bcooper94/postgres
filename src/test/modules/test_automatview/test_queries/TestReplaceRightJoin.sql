select employee.id, person_id, employee_id from employee join manager on employee.id = employee_id;
select employee.id, person_id from employee join manager on employee.id = employee_id;

select person_id, name, age from person join employee on person.id = person_id join manager on employee.id = employee_id;

/* Expected:
 * 
 * select MatView.employee_person_id, person.name, person.age
 * 	from person join MatView on person.id = MatView.employee_person_id
 */

/* Actual (3/18/18):
 * SELECT Matview.employee_person_id, person.name, person.age
 * FROM person JOIN Matview ON person.id = Matview.employee_person_id
 */