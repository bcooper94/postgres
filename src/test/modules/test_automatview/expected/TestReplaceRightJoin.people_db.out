1
SELECT Matview.employee_person_id AS Matview_employee_person_id, person.name AS person_name, person.age AS person_age FROM person INNER JOIN Matview ON person.id = Matview.employee_person_id